src/pages/LogInWithShortLivedTokenPage.js

```diff
@@ -38,37 +39,58 @@ const propTypes = {
         const email = lodashGet(this.props.route.params, 'email', '');
         const shortLivedToken = lodashGet(this.props.route.params, 'shortLivedToken', '');
         const encryptedAuthToken = lodashGet(this.props.route.params, 'encryptedAuthToken', '');
 
-        // exitTo is URI encoded because it could contain a variable number of slashes (i.e. "workspace/new" vs "workspace/<ID>/card")
-        const exitTo = decodeURIComponent(lodashGet(this.props.route.params, 'exitTo', ''));
-
-        // If the user is revisiting the component authenticated or they were already logged into the right account, we simply redirect them to the exitTo
+        // If the user is revisiting the component authenticated or they were already logged into
+        // the right account, we need to ensure they have the betas loaded, then in componentDidUpdate we will
         if (this.props.session.authToken && email === this.props.session.email) {
-            // In order to navigate to a modal, we first have to dismiss the current modal. But there is no current
-            // modal you say? I know, it confuses me too. Without dismissing the current modal, if the user cancels out
-            // of the workspace modal, then they will be routed back to
-            // /transition/<accountID>/<email>/<authToken>/workspace/<policyID>/card and we don't want that. We want them to go back to `/`
-            // and by calling dismissModal(), the /transition/... route is removed from history so the user will get taken to `/`
-            // if they cancel out of the new workspace modal.
-            Navigation.dismissModal();
-            Navigation.navigate(exitTo);
+            User.getBetas();
+            return;
         }
 
         signInWithShortLivedToken(accountID, email, shortLivedToken, encryptedAuthToken);
+        this.setState({run: true});
+    }
+
+    componentDidUpdate() {
+        if (this.state.run || !this.props.betas) {
+            return;
+        }
+
+        // exitTo is URI encoded because it could contain a variable number of slashes (i.e. "workspace/new" vs "workspace/<ID>/card")
+        const exitTo = decodeURIComponent(lodashGet(this.props.route.params, 'exitTo', ''));
+
+        // In order to navigate to a modal, we first have to dismiss the current modal. But there is no current
+        // modal you say? I know, it confuses me too. Without dismissing the current modal, if the user cancels out
+        // of the workspace modal, then they will be routed back to
+        // /transition/<accountID>/<email>/<authToken>/workspace/<policyID>/card and we don't want that. We want them to go back to `/`
+        // and by calling dismissModal(), the /transition/... route is removed from history so the user will get taken to `/`
+        // if they cancel out of the new workspace modal.
+        Navigation.dismissModal();
+        Navigation.navigate(exitTo);
```

Comment:
Final comment, do we need to call `this.setState({run: true})` here too, to prevent other GetBeta API calls from triggering `Navigation.navigate` a second time?

Reply:
As the GetBetas API call is now happening twice

Reply:
Fixed that